pkg/controllers/node/expiration.go

```diff
@@ -0,0 +1,58 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha3"
+	"github.com/awslabs/karpenter/pkg/utils/ptr"
+	v1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/logging"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// Expiration is a tiny reconciler that ensures nodes have the termination
+// Expiration. This protects against instances that launch when Karpenter fails
+// to create the node object. In this case, the node will come online without
+// the termination Expiration. This controller will update the node accordingly.
+type Expiration struct {
+	kubeClient client.Client
+}
+
+// Reconcile adds the termination Expiration if the node is not deleting
```

Comment:
This comment doesn't make much sense. Could you say "Reconcile deletes the node if it is past its expiration date"?